code/modules/antagonists/heretic/knowledge/rust_lore.dm

```diff
@@ -66,13 +66,15 @@
 
 /datum/heretic_knowledge/rust_fist/proc/on_secondary_mansus_grasp(mob/living/source, atom/target)
 	SIGNAL_HANDLER
-
+	for(var/dir in GLOB.cardinals)
+		var/turf/nearby_turf = get_step(target, dir)
+		if(istype(nearby_turf))
+			nearby_turf.rust_heretic_act()
```

Comment:
You should only be rusting adjacent turfs if it's being cast on a turf

In the current implementation, if I use rust grasp to break down an airlock, it'll leave behind a huge patch of rust even when I only wanted to hit the airlock